### PR TITLE
New chat client / messenger

### PR DESCRIPTION
**Pull request by Karolis2011**
Targets to refactor messenger, expands it's functionality makes it less copy pasty and also new vueui is comming.

- [x] New vueui ui for the interface [image](https://user-images.githubusercontent.com/11229891/100524959-6c496600-31c5-11eb-9faf-4af55e5851ee.png)
- [x] Removes logging and username changing functions
- [x] Allow multiple devices to use same username
- [x] Rewrititng of most of backend.
- [x] Allow user kicking from channels.
- [x] Fix channel deletion.
- [x] Chat log has been reversed.
- [x] Test it out.

### CODE REVIEW

**Comment by Karolis2011** [comment]
!review

**Comment by skull132** [comment]
Reported bugs:

- Viewing lists (or any other actually formatted fields) in VV is broken.
- Messenger on the laptop may not work? Single report, couldn't confirm.

File: code/modules/modular_computers/computers/subtypes/dev_silicon.dm [hunk 1]
```diff
@@ -62,23 +62,23 @@
...
-	if (usr.stat == DEAD)
-		to_chat(usr, "You can't send PDA messages because you are dead!")
-		return
-	var/datum/computer_file/program/chatclient/CL = hard_drive.find_file_by_name("ntnrc_client")
-	if(!istype(CL))
-		output_error("Chat client not installed!")
-		return
-	else if(CL.program_state == PROGRAM_STATE_KILLED)
-		run_program("ntnrc_client")
+// /obj/item/modular_computer/silicon/verb/send_pda_message()
```
Comment (Geevies) [inline code/modules/modular_computers/computers/subtypes/dev_silicon.dm:65]: dead code, can remove

**Review by Geevies — APPROVED** [review]
looks about right, i don't spot any glaring errors

**Review by skull132 — APPROVED** [review]
This was test merged and worked Fine:tm:.

**Pull request merged**